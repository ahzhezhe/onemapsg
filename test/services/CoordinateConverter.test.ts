import { oneMap } from '..';

test('_4326to3857', async () => {
  const result = await oneMap.coordinateConverter._4326to3857({
    latitude: 1.3039454,
    longitude: 103.8408471
  });
  expect(result).toMatchSnapshot();
});

test('_4326to3414', async () => {
  const result = await oneMap.coordinateConverter._4326to3414({
    latitude: 1.3039454,
    longitude: 103.8408471
  });
  expect(result).toMatchSnapshot();
});

test('_3414to3857', async () => {
  const result = await oneMap.coordinateConverter._3414to3857({
    X: 28833.0605447,
    Y: 31810.3242714
  });
  expect(result).toMatchSnapshot();
});

test('_3414to4326', async () => {
  const result = await oneMap.coordinateConverter._3414to4326({
    X: 28833.0605447,
    Y: 31810.3242714
  });
  expect(result).toMatchSnapshot();
});

test('_3857to3414', async () => {
  const result = await oneMap.coordinateConverter._3857to3414({
    X: 28833.0605447,
    Y: 31810.3242714
  });
  expect(result).toMatchSnapshot();
});

test('_3857to4326', async () => {
  const result = await oneMap.coordinateConverter._3857to4326({
    X: 28833.0605447,
    Y: 31810.3242714
  });
  expect(result).toMatchSnapshot();
});
