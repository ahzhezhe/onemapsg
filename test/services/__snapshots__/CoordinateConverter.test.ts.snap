// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_3414to3857 1`] = `
{
  "X": 11559505.426495155,
  "Y": 145168.22515712655,
}
`;

exports[`_3414to4326 1`] = `
{
  "latitude": 1.303955777623504,
  "longitude": 103.84080401483146,
}
`;

exports[`_3857to3414 1`] = `
{
  "X": 28837.8555334731,
  "Y": 31809.17678148796,
}
`;

exports[`_3857to4326 1`] = `
{
  "latitude": 1.3039454000000008,
  "longitude": 103.8408471,
}
`;

exports[`_4326to3414 1`] = `
{
  "X": 28837.8555334731,
  "Y": 31809.17678148787,
}
`;

exports[`_4326to3857 1`] = `
{
  "X": 11559510.222714178,
  "Y": 145167.06962612906,
}
`;
